### Heroes Of Pymoli Data Analysis
* Of the 1163 active players, the vast majority are male (84%). There also exists, a smaller, but notable proportion of female players (14%).

* Our peak age demographic falls between 20-24 (44.8%) with secondary groups falling between 15-19 (18.60%) and 25-29 (13.4%).  
-----

### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [1]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)


## Player Count

In [2]:
print("The total number of players is",len(purchase_data["SN"].value_counts()))


The total number of players is 576


## Purchasing Analysis (Total)

In [3]:
unique_id=purchase_data["Item ID"].unique()

average_price=((purchase_data["Price"]).sum())/(purchase_data["SN"].count())
average_price=average_price.round(decimals=2)

total_purchase=purchase_data["Purchase ID"].value_counts()

total_revenue=purchase_data["Price"].sum()

analysis=pd.DataFrame({
    "Unique Items":[len(unique_id)],
    "Avg Price of Purchase($)":[average_price],
    "Total Purchases":[len(total_purchase)],
    "Total Revenue($)":[total_revenue]
})
analysis

Unique Items  Avg Price of Purchase($)  Total Purchases  Total Revenue($)
0           183                      3.05              780           2379.77

## Gender Demographics

In [11]:
a=purchase_data.groupby(["Gender"]).agg({
    "SN":"nunique"
})
a["Percent"]=(a["SN"]/576)*100
a.columns=["Count","Percent"]
a["Percent"]=a["Percent"].round()
a



Count  Percent
Gender                               
Female                    81     14.0
Male                     484     84.0
Other / Non-Disclosed     11      2.0


## Purchasing Analysis (Gender)

In [15]:
gender_summary=purchase_data.groupby("Gender").agg({
    "Purchase ID":'count',
    "Price":["mean","sum"],
    "SN":"nunique"})
gender_summary.columns=[' '.join(col).strip() for col in gender_summary.columns.values]
gender_summary["Avg Total per Person"]=(gender_summary["Price sum"]/gender_summary["SN nunique"])
gender_summary["Price mean"]=gender_summary["Price mean"].round(decimals=2)
gender_summary["Avg Total per Person"]=gender_summary["Avg Total per Person"].round(decimals=2)
gender_summary[["Purchase ID count","Price mean","Price sum","Avg Total per Person"]]



Purchase ID count  Price mean  Price sum  \
Gender                                                            
Female                               113        3.20     361.94   
Male                                 652        3.02    1967.64   
Other / Non-Disclosed                 15        3.35      50.19   

                       Avg Total per Person  
Gender                                       
Female                                 4.47  
Male                                   4.07  
Other / Non-Disclosed                  4.56

## Age Demographics

In [12]:
bins=[0,9,14,19,24,29,34,39,44,49]
labels=["0-9","10-14","15-19","20-24","25-29","30-34","35-39","40-44","45-49"]
pd.cut(purchase_data["Age"],bins,labels=labels).head()
purchase_data["Age Range"]=pd.cut(purchase_data["Age"],bins,labels=labels)

b=purchase_data.groupby(["Age Range"]).agg({
    "SN":"nunique"
})
b["Percent"]=(b["SN"]/576)*100
b.columns=["Count","Percent"]
b["Percent"]=b["Percent"].round()
b

Count  Percent
Age Range                
0-9           17      3.0
10-14         22      4.0
15-19        107     19.0
20-24        258     45.0
25-29         77     13.0
30-34         52      9.0
35-39         31      5.0
40-44         11      2.0
45-49          1      0.0

## Purchasing Analysis (Age)

In [17]:
age_summary=purchase_data.groupby("Age Range").agg({
    "Purchase ID":'count',
    "Price":["mean","sum"],
    "SN":"nunique"})
age_summary.columns=[' '.join(col).strip() for col in age_summary.columns.values]
age_summary["Avg Total per Person"]=(age_summary["Price sum"]/age_summary["SN nunique"])
age_summary["Price mean"]=age_summary["Price mean"].round(decimals=2)
age_summary["Avg Total per Person"]=age_summary["Avg Total per Person"].round(decimals=2)
age_summary[["Purchase ID count","Price mean","Price sum","Avg Total per Person"]]





Purchase ID count  Price mean  Price sum  Avg Total per Person
Age Range                                                                
0-9                       23        3.35      77.13                  4.54
10-14                     28        2.96      82.78                  3.76
15-19                    136        3.04     412.89                  3.86
20-24                    365        3.05    1114.06                  4.32
25-29                    101        2.90     293.00                  3.81
30-34                     73        2.93     214.00                  4.12
35-39                     41        3.60     147.67                  4.76
40-44                     12        3.04      36.54                  3.32
45-49                      1        1.70       1.70                  1.70

## Top Spenders

In [18]:
summary=purchase_data.groupby("SN").agg(
    {'Price':['sum','mean'],
    'Purchase ID':'count'})
summary.columns=[' '.join(col).strip() for col in summary.columns.values]
summary["Price mean"]=summary["Price mean"].round(decimals=2)
summary.sort_values(by='Price sum',ascending=False).head()



Price sum  Price mean  Purchase ID count
SN                                                   
Lisosia93        18.96        3.79                  5
Idastidru52      15.45        3.86                  4
Chamjask73       13.83        4.61                  3
Iral74           13.62        3.40                  4
Iskadarya95      13.10        4.37                  3

## Most Popular Items

In [9]:

popular=purchase_data.groupby(["Item Name","Item ID"]).agg({
    "Purchase ID":"count",
    "Price":["sum","min"]
})
popular.columns=[' '.join(col).strip() for col in popular.columns.values]
popular.sort_values(by='Purchase ID count',ascending=False).head()

Purchase ID count  \
Item Name                                    Item ID                      
Oathbreaker, Last Hope of the Breaking Storm 178                     12   
Extraction, Quickblade Of Trembling Hands    108                      9   
Nirvana                                      82                       9   
Fiery Glass Crusader                         145                      9   
Pursuit, Cudgel of Necromancy                19                       8   

                                                      Price sum  Price min  
Item Name                                    Item ID                        
Oathbreaker, Last Hope of the Breaking Storm 178          50.76       4.23  
Extraction, Quickblade Of Trembling Hands    108          31.77       3.53  
Nirvana                                      82           44.10       4.90  
Fiery Glass Crusader                         145          41.22       4.58  
Pursuit, Cudgel of Necromancy                19            8.16       1.02

## Most Profitable Items

In [10]:
profitable=purchase_data.groupby(["Item Name","Item ID"]).agg({
    "Purchase ID":"count",
    "Price":["sum","min"]
})
profitable.columns=[' '.join(col).strip() for col in profitable.columns.values]
profitable.sort_values(by='Price sum',ascending=False).head()

Purchase ID count  \
Item Name                                    Item ID                      
Oathbreaker, Last Hope of the Breaking Storm 178                     12   
Nirvana                                      82                       9   
Fiery Glass Crusader                         145                      9   
Final Critic                                 92                       8   
Singed Scalpel                               103                      8   

                                                      Price sum  Price min  
Item Name                                    Item ID                        
Oathbreaker, Last Hope of the Breaking Storm 178          50.76       4.23  
Nirvana                                      82           44.10       4.90  
Fiery Glass Crusader                         145          41.22       4.58  
Final Critic                                 92           39.04       4.88  
Singed Scalpel                               103          34.80       4.35

## FINAL ANALYSIS

1.The most common age range for game purchasing is 20-24 years old, with 258 gamers. As a group, this age range spent over $600 dollars more than any other age range on games.

2.Males make up the vast majority of purchasers, with 484 male individuals making up 84 percent of purchasers. As a group men have spent the most money on games than any other gender, but have the lowest average purchase price.

3.All of the most profitable games have a price above $4. Three of the top five most profitable games also appear in the most popular games top five, but the other two games in the top five most popular both have prices below four dollars and do not make into the most profitable games.